lib/internal/Magento/Framework/Escaper.php

```diff
@@ -45,6 +45,35 @@ public function escapeHtml($data, $allowedTags = null)
     }
 
     /**
+     * Escape XML entities
+     *
+     * @param  string|array $data
+     * @param  array $allowedTags
+     * @return string|array
+     */
+    public function escapeXml($data, $allowedTags = null)
```

Comment:
Taking into account backward compatibility requirements, we can't add public methods to classes which could be considered as Magento APIs, because potentially it could lead to conflicts if this class has been extended by 3rd party developer, and new method with the same name has been added in child class.
So, it's better to introduce own XmlEscaper class with logic you added.

And inject an instance of this new entity into the app/code/Magento/Widget/Model/Widget/Instance as optional dependency (default value is equal to NULL)